### ISSUE TITLE
Investigate use of "filetype" module for output_checkers "correct_file_format" function

### ISSUE THREAD

**Issue by evrose54** [NRLMMD-GEOIPS/geoips#372]
# Requested Update

### Description
Currently, the correct_file_format function uses the extension of the provided file to determine whether or not the file provided is correct for a certain output_checker. We should investigate whether or not the "filetype" module is better suited for determining if the file provided is correct for a certain output_checker.

### Background and Motivation
It's possible that a user could mislabel a file with an incorrect extension, and this module provides functionality to circumvent that case.

### Alternative Solutions
Leave it as is, and raise an error if the file provided doesn't have attributes needed for certain comparisons.

**Comment by evrose54** [comment]
<img width="737" alt="Screenshot 2023-10-13 at 2 52 27 PM" src="https://github.com/NRLMMD-GEOIPS/geoips/assets/62626811/63cac5e7-925e-4033-9857-cf1ec6cc91fb">

**Comment by evrose54** [comment]
Looks like this would work. After viewing their [GitHub Page](https://github.com/h2non/filetype.py), they state that this package "Small and dependency free [Python](http://python.org/) package to infer file type and MIME type checking the [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)#Magic_numbers_in_files) signature of a file or buffer."

This is a Python port from [filetype](https://github.com/h2non/filetype) Go package.

**Comment by evrose54** [comment]
@jsolbrig, if you like this, I can try implementing this in a future PR.

**Comment by jsolbrig** [comment]
@evrose54 Yeah, I think we should do this as long as @mindyls is okay with it. It would allow checking both based on extension and magic number. I do wonder, though, which takes precedence. For example:

1. If you had a NetCDF file named `myfile.nc` and renamed it to `myfile.txt`, what would get reported?
2. Alternatively, what if you had a text file named `myfile.txt` and renamed it to `myfile.nc`? The text file doesn't contain a magic number, but a NetCDF file should contain one. Maybe this would raise an error?

**Comment by mindyls** [comment]
I think this is fine, but pretty low priority from my perspective.  We can address the magic number thing when we begin implementing.  We definitely want to wait to implement this until we have our unit tests under control.

**Comment by jsolbrig** [comment]
I think we would like to do this now as part of the updates that allow not setting an output checker in the `test` section of our OBP workflows.

This should be implemented such that, at expansion time, prior to insertion into the spec section, we would automatically determine the correct output_checker to use, then insert it as appropriate. I'll ask @evrose54 and @coleman-m to collaborate on this.
